src/components/CvssCalculator/CvssCalculatorBase.vue

```diff
@@ -0,0 +1,189 @@
+<script setup lang="ts">
+import { computed, onMounted, ref, toRaw, watch } from 'vue';
+
+import { equals } from 'ramda';
+
+import { useCvssScores, validateCvssVector } from '@/composables/useCvssScores';
+import { calculateCvss3Score, parseCvss3Factors } from '@/composables/useCvss3Calculator';
+
+import type { CvssEntity, ZodAffectCVSSType } from '@/types';
+import { CvssVersions } from '@/constants';
+import { deepCopyFromRaw } from '@/utils/helpers';
+
+import Cvss3Calculator from './Cvss3Calculator/Cvss3Calculator.vue';
+import CvssVectorInput from './CvssVectorInput.vue';
+
+const props = defineProps<{
+  cvssEntity: CvssEntity;
+}>();
+
+const emit = defineEmits<{
+  'blur': [FocusEvent];
+  'change:cvss_score': [ZodAffectCVSSType[]];
+}>();
+
+const highlightedFactor = ref<null | string>(null);
+const highlightedFactorValue = ref<null | string>(null);
+const cvssDiv = ref();
+const isFocused = ref(false);
+const cvssVectorInput = ref();
+const cvssEntity = ref<CvssEntity>(deepCopyFromRaw(toRaw(props.cvssEntity)));
+
+const {
+  cvss3Factors,
+  cvssScore,
+  cvssVector,
+  cvssVersion,
+  updateScore,
+  updateVector,
+} = useCvssScores(cvssEntity.value);
+
+const error = computed(() => validateCvssVector(cvssVector.value, cvssVersion.value) ?? null);
+
+watch(() => cvssEntity.value, () => {
+  if (!equals(props.cvssEntity.cvss_scores, cvssEntity.value.cvss_scores)) {
+    emit('change:cvss_score', cvssEntity.value.cvss_scores);
+  }
+}, { deep: true });
+
+watch(
+  () => cvssVector.value,
+  () => {
+    updateUsingV3Vector(cvssVector.value);
+    updateCvss(cvssVector.value);
+  },
+  { immediate: true },
+);
+
+function onInputBlur(event: FocusEvent) {
+  if (event.relatedTarget !== cvssDiv.value) {
+    isFocused.value = false;
+    emit('blur', event);
+  }
+}
+
+function onInputFocus(event: FocusEvent) {
+  isFocused.value = true;
+  if (event.target !== cvssVectorInput.value.input) {
+    cvssVectorInput.value.input.focus();
+  }
+}
+
+function reset() {
+  updateScore(null);
+  updateVector(null);
+  cvss3Factors.value = {};
+}
```

Comment:
`reset` seems more at home in the CVSS composable, but I suspect there's a reason it was factored out

Reply:
I didn't refactor things in/out of the composable, only moved the logic from the `Overlayed` to the `Base`, so this was already like that